src/main/java/org/junit/experimental/categories/Categories.java

```diff
@@ -346,11 +347,28 @@ private static boolean hasAssignableTo(Set<Class<?>> assigns, Class<?> to) {
         return false;
     }
 
-    private static Set<Class<?>> createSet(Class<?>... t) {
-        final Set<Class<?>> set= new HashSet<Class<?>>();
-        if (t != null) {
-            Collections.addAll(set, t);
+    private static Set<Class<?>> createSet(Class<?>[] t) {
+        // Not throwing a NPE if t is null is a bad idea, but it's the behavior from JUnit 4.12
+        // for include(boolean, Class<?>...) and exclude(boolean, Class<?>...)
+        if (t == null || t.length == 0) {
+            return Collections.emptySet();
+        }
+        for (Class<?> category : t) {
+            if (category == null) {
+                throw new NullPointerException("has null category");
+            }
         }
-        return set;
+
+        return t.length == 1
+            ? Collections.<Class<?>>singleton(t[0])
+            : new HashSet<Class<?>>(Arrays.asList(t));
```

Comment:
Good point about debugging. The `toString()` and `describe()` methods do print out the categories. Printing them in the same order makes sense. Changed to use `LinkedHashSet` in a separate commit